plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,564 @@
+import org.apache.iceberg.ReplacePartitions;
+import org.apache.iceberg.ReplaceSortOrder;
+import org.apache.iceberg.RewriteFiles;
+import org.apache.iceberg.RewriteManifests;
+import org.apache.iceberg.Rollback;
+import org.apache.iceberg.RowDelta;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.UpdateLocation;
+import org.apache.iceberg.UpdatePartitionSpec;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.plugin.iceberg.TrinoHiveCatalog.DEPENDS_ON_TABLES;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+public class TestIcebergMetadata
+{
+    private static final CatalogSchemaTableName TESTING_MATERIALIZED_VIEW_NAME = new CatalogSchemaTableName(
+            "test_catalog_name",
+            "test_schema_name",
+            "test_table_name");
+    private static final ConnectorMaterializedViewDefinition TEST_MATERIALIZED_VIEW_DEFINITION = new ConnectorMaterializedViewDefinition(
+            "originalSql",
+            Optional.of(TESTING_MATERIALIZED_VIEW_NAME),
+            Optional.empty(),
+            Optional.empty(),
+            ImmutableList.of(new Column("test_column", TypeId.of("double"))),
+            Optional.empty(),
+            "",
+            ImmutableMap.of());
+
+    private IcebergMetadata testingIcebergMetadata;
+    private TestingTrinoCatalog testingTrinoCatalog;
+
+    @BeforeClass
+    public void setup()
+    {
+        testingTrinoCatalog = new TestingTrinoCatalog();
+        testingIcebergMetadata = new IcebergMetadata(new TestingTypeManager(), jsonCodec(CommitTaskData.class), testingTrinoCatalog);
+    }
+
+    @Test
+    public void testGetMaterializedViewFreshness()
+    {
+        testingTrinoCatalog.returnWhenGetMaterializedView(Optional.of(TEST_MATERIALIZED_VIEW_DEFINITION));
+
+        assertThatThrownBy(() ->
+                testingIcebergMetadata.getMaterializedViewFreshness(
+                        TestingConnectorSession.SESSION,
+                        TESTING_MATERIALIZED_VIEW_NAME.getSchemaTableName()))
+                .isInstanceOf(MaterializedViewNotFoundException.class)
+                .hasMessage(format("Materialized View '%s' not found", TESTING_MATERIALIZED_VIEW_NAME.getSchemaTableName()));
```

Comment:
why does this throw an exception since you called `testingTrinoCatalog.returnWhenGetMaterializedView(Optional.of(TEST_MATERIALIZED_VIEW_DEFINITION));` before?

Reply:
That would mean when `getMaterializedViewFreshness` -> `getMaterializedViewToken ` ->  `getMaterializedView` is stubbed:

https://github.com/trinodb/trino/blob/6f1297f3fc60671873b2f59e742b03e9c0c2195b/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L872

`getTableHandle` is the method which will return null and cause exception to be thrown:

https://github.com/trinodb/trino/blob/6f1297f3fc60671873b2f59e742b03e9c0c2195b/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L862

It's a bit similar to Mockito (ex: `Mockito.when(someDependency.getThing()).thenReturn("Some Other thing");`)

Reply:
But I removed this class anyways.